### PR TITLE
Ankit | Test | Provide support of multiple deposit in single voucher in create/update flow (PAID)

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cvwurn7

### CODE REVIEW

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.dialog-body .deposit-wrapper {
+    max-height:194px;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.scss]: format in scss

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 2]
```diff
@@ -193,8 +228,17 @@ export class PaymentDialogComponent implements OnInit, OnDestroy {
      * @param {string} currencySymbol
      * @memberof PaymentDialogComponent
      */
-    private assignAmount(amount: number, currencySymbol: string): void {
-        this.paymentForm.get('amount')?.patchValue(amount);
+    private assignAmount(amount: number, currencySymbol: string, indexValue: number = 0): void {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts]:
```suggestion
    private assignAmount(amount: number, currencySymbol: string, depositIndex: number = 0): void {
```
make changes in overall file

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 3]
```diff
@@ -236,16 +280,20 @@ export class PaymentDialogComponent implements OnInit, OnDestroy {
         }
 
         if (this.voucherDetails?.account?.currency?.code === this.selectedPaymentMode?.additional?.currency) {
-            newFormObj.amountForAccount = newFormObj.amount;
+            newFormObj.deposits.forEach((deposit) => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts]:
```suggestion
            newFormObj.deposits.forEach((deposit:any) => {
```

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 4]
```diff
@@ -236,16 +280,20 @@ export class PaymentDialogComponent implements OnInit, OnDestroy {
...
+                deposit.amountForAccount = deposit.amount;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts]:
```suggestion
                deposit.amountForAccount = deposit?.amount;
```

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 5]
```diff
@@ -236,16 +280,20 @@ export class PaymentDialogComponent implements OnInit, OnDestroy {
...
+                delete deposit.amount;
+            });
         } else {
-            newFormObj.amountForCompany = newFormObj.amount;
+            newFormObj.deposits.forEach((deposit) => {
+                deposit.amountForCompany = deposit.amount;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts:289]: remove from if else and confirm with @dvCodeWorld

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 6]
```diff
@@ -193,8 +228,17 @@ export class PaymentDialogComponent implements OnInit, OnDestroy {
...
      * @memberof PaymentDialogComponent
      */
-    private assignAmount(amount: number, currencySymbol: string): void {
-        this.paymentForm.get('amount')?.patchValue(amount);
+    private assignAmount(amount: number, currencySymbol: string, indexValue: number = 0): void {
+        this.paymentForm.get('deposits')['controls']?.forEach((control: any, index: number) => {
+            if (control.get('amount').value && indexValue !== index) {
+                amount -= Number(control.get('amount').value);
+            }
+        });
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts:236]: confirm this logic with Kriti

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.scss (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.scss]: Done

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts]: Done

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts]: Done

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.html [hunk 7]
```diff
@@ -4,68 +4,144 @@
...
+            formControlName="tagUniqueName"
+        >
         </reactive-dropdown-field>
     </div>
+
+    <div class="container px-0 pd-t05 deposit-wrapper overflow-x-hidden overflow-y-auto" formArrayName="deposits">
+        <ng-container *ngFor="let deposit of paymentForm?.get('deposits')['controls']; let depositIndex = index">
+            <div class="d-flex column-gap1 mr-b15" [formGroupName]="depositIndex">
+                <reactive-dropdown-field
+                    [label]="commonLocaleData?.app_payment_mode + '*'"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.html]:
```suggestion
                    [label]="commonLocaleData?.app_payment_mode"
```

[required]="true"

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**